src/rust/rust-proto/src/client_macros.rs

```diff
@@ -0,0 +1,38 @@
+/// This macro implements boilerplate code to:
+/// - translate between the native types and tonic/prost types
+///   in the transport layer
+/// - Automatically hook these client RPCs up to the Executor, which provides
+///   retries (among other good behaviors).
+///
+/// It makes some expectations of the structure of `self` - like that it
+/// has a `self.executor` and `self.proto_client`.
+#[macro_export]
+macro_rules! execute_client_rpc {
```

Comment:
Can we call this `execute_rpc` instead? The name `execute_client_rpc` sounds like it's executing a remote procedure call _against the client_ which doesn't really make sense. Instead, we're executing a remote procedure call against the server.